### ISSUE TITLE
Improve autoupdate download speed and stability

### ISSUE THREAD

**Issue by tsusanka** [trezor/trezor-suite#4692]
As some noticed the download speed during autoupdate is somewhat unsatisfactory. Most likely (though I haven't found clear numbers) Github throttles the download speed. It would be great to download binaries directly from our S3 bucket instead of Github.

Note that we if agree upon https://github.com/trezor/webwallet-data/issues/34 we can simply replace the github link with data.trezor.io (see the issue for details) and that's it. I hope that is easy to do in the autoupdater.

**Comment by matejkriz** [comment]
Also there are lot of network issues in Sentry related to github releases so it's not only slow, but also unreliable...

**Comment by matejkriz** [comment]
related to https://github.com/trezor/trezor-suite/issues/5214

**Comment by tsusanka** [comment]
Please note that as @matejkriz pointed out elsewhere, this is not just a cosmetic product improvement, it will also improve the autoupdate's reliability, as we are seeing quite some autoupdate errors in Sentry.

**Comment by tsusanka** [comment]
A summary after discussing this with @matejkriz:

1. We see some errors in Sentry related to the autoupdate but we do not know for certain that it would be because of some GitHub limits or something similar.
2. Simple AWS Cloudfront caching will not do the trick because you can not simply replace URL in the GitHub provider in the Electron autoupdate. It is tied to github.com. Pity, it would be a nice solution.
3. The solution introduced for QA is using a "generic" provider in the Electron autoupdate and is separated from the GitHub workflow. We could use that but it brings more complexity, for example find out where we will fetch the changelog from etc.

----

Tl;dr: we first need to find out what exactly is a problem here and then suggest a solution. So let's first assess the priority properly. As it comes from Sentry I am marking it as _code_, and let's triage it with devs first.

**Comment by matejkriz** [comment]
Interesting issue to read the comments https://github.com/electron-userland/electron-builder/issues/7398

**Comment by matejkriz** [comment]
Switching auto-update provider could also solve https://github.com/trezor/trezor-suite/issues/12236
